shell/platform/embedder/embedder_external_texture_metal.mm

```diff
@@ -0,0 +1,97 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/embedder/embedder_external_texture_metal.h"
+
+#include "flutter/fml/logging.h"
+#import "flutter/shell/platform/darwin/graphics/FlutterDarwinExternalTextureMetal.h"
+#include "third_party/skia/include/core/SkImage.h"
+#include "third_party/skia/include/core/SkSize.h"
+#include "third_party/skia/include/gpu/GrBackendSurface.h"
+#include "third_party/skia/include/gpu/GrDirectContext.h"
+
+namespace flutter {
+
+EmbedderExternalTextureMetal::EmbedderExternalTextureMetal(int64_t texture_identifier,
+                                                           const ExternalTextureCallback& callback)
+    : Texture(texture_identifier), external_texture_callback_(callback) {
+  FML_DCHECK(external_texture_callback_);
+}
+
+EmbedderExternalTextureMetal::~EmbedderExternalTextureMetal() = default;
+
+// |flutter::Texture|
+void EmbedderExternalTextureMetal::Paint(SkCanvas& canvas,
+                                         const SkRect& bounds,
+                                         bool freeze,
+                                         GrDirectContext* context,
+                                         const SkSamplingOptions& sampling) {
+  if (auto image = ResolveTexture(Id(), context, SkISize::Make(bounds.width(), bounds.height()))) {
+    last_image_ = image;
+  }
+
+  if (last_image_) {
+    if (bounds != SkRect::Make(last_image_->bounds())) {
+      canvas.drawImageRect(last_image_, bounds, sampling);
+    } else {
+      canvas.drawImage(last_image_, bounds.x(), bounds.y(), sampling, nullptr);
+    }
+  }
+}
+
+sk_sp<SkImage> EmbedderExternalTextureMetal::ResolveTexture(int64_t texture_id,
+                                                            GrDirectContext* context,
+                                                            const SkISize& size) {
+  std::unique_ptr<FlutterMetalExternalTexture> texture =
+      external_texture_callback_(texture_id, size.width(), size.height());
+
+  if (!texture) {
+    return nullptr;
+  }
+
+  sk_sp<SkImage> image;
+
+  switch (texture->pixel_format) {
+    case FlutterMetalExternalTexturePixelFormat::kRGBA: {
+      FML_CHECK(texture->num_textures == 1);
```

Comment:
This should be a recoverable error. You could log and/or return the error instead of terminating the process.

Reply:
How about just logging something like "Incorrect number of textures specified for Formate::kRGBA" and displaying a blank image.